### ISSUE TITLE
Is there a way to build files on git pull?

### ISSUE THREAD

**Issue by blakeembrey** [keymetrics/pm2-auto-pull#2]
For example, my module is not written directly in JavaScript and it has to compile first. Although it pulls the changes, it can't actually use them until I build it.

**Comment by AkarshSatija** [comment]
This isn't actually a hook... it checks and pulls after an interval
You can use hooks and configure it for you to run a particular process

**Comment by blakeembrey** [comment]
@akarshsatija I know this isn't a hook, but I guess you're talking about http://pm2.keymetrics.io/docs/usage/deployment/ then? Do you have a link to the hooks you're talking about?

**Comment by AkarshSatija** [comment]
@blakeembrey Why don't you use github webHooks, can help you triggering anything.
https://developer.github.com/webhooks/

**Comment by blakeembrey** [comment]
@akarshsatija I'm honestly not sure how that would help me. I'm trying to recompile the application running with PM2 since the built files are not version controlled.

**Comment by AkarshSatija** [comment]
Practically you can trigger any process on a server with this, but thats a tedious task to setup commands on it.
So, Would suggest you to setup a task manager like GRUNT with watch files
Ref- https://github.com/gruntjs/grunt
https://github.com/gruntjs/grunt-contrib-watch

**Comment by blakeembrey** [comment]
@akarshsatija Why would I want to set up a separate watch task from my application when this module already has all the information I need to know when to recompile assets? It would just be additional overhead with negative gain.

**Comment by ostrgard** [comment]
@blakeembrey, you're essentially asking for the post-deploy specified in ecosystem.json to be executed, right? As far as I can understand, this is not happening as of right now, is it?

Would it be possible?

**Comment by blakeembrey** [comment]
I think that was one way to do it. However, being new to PM2 the way I deployed the script was literally git pull and then I wanted to add this. Since it's a TypeScript project though, I need to run a command to compile before it reloads - I wouldn't really mind how, could be a field in `package.json` if anything.

**Comment by AngeloR** [comment]
Is relying on git hooks (instead of github webhooks) out of the question? That way on your deployment server, you have a git post-receive hook configured locally. After the "git pull" happens from pm2, you run the TS compiler and then pm2 restarts.

**Comment by blakeembrey** [comment]
Nope, that would certainly work. My only question from there is, could this be easier? What about running the `npm prepublish` or `npm install` too?

**Comment by AngeloR** [comment]
I'm more inclined to have it as apart of your npm scripts - so that it's versioned and available to all contributors. There's some more information on using npm as a build tool here: http://blog.keithcirkel.co.uk/how-to-use-npm-as-a-build-tool/

**Comment by blakeembrey** [comment]
@AngeloR Yes, I do that already, but I still need a way for this to execute it on "auto-pull".

**Comment by mathew-kurian** [comment]
Posted this here before, but here a PM2 plugin for that https://github.com/bluejamesbond/BranchOff.js

**Comment by sladec** [comment]
I need "git pull" ->  "npm install" -> PM2 restart ?
Cant believe this is not an issue for many when you add new packages to package.json ?

**Comment by rakshith-ravi** [comment]
Is there any update to this yet?
